program/rust/src/accounts/price.rs

```diff
@@ -91,3 +94,101 @@ impl PythAccount for PriceAccount {
     /// Equal to the offset of `comp_` in `PriceAccount`, see the trait comment for more detail
     const INITIAL_SIZE: u32 = PC_PRICE_T_COMP_OFFSET as u32;
 }
+
+/// Message format for sending data to other chains via the accumulator program
+/// When serialized, each message starts with a unique 1-byte discriminator, followed by the
+/// serialized struct data in the definition(s) below.
+///
+/// Messages are forward-compatible. You may add new fields to messages after all previously
+/// defined fields. All code for parsing messages must ignore any extraneous bytes at the end of
+/// the message (which could be fields that the code does not yet understand).
+#[repr(C)]
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct PriceFeedMessage {
+    pub id:                [u8; 32],
+    pub price:             i64,
+    pub conf:              u64,
+    pub exponent:          i32,
+    /// The timestamp of this price update
+    pub publish_time:      i64,
+    /// The timestamp of the previous price update. This field is intended to allow users to
+    /// identify the single unique price update immediately after any moment in time:
+    /// for any time t, the unique update is the one such that prev_publish_time <= t < publish_time.
+    ///
+    /// Note that there may not be such an update while we are migrating to the new message-sending logic,
+    /// as some price updates on pythnet may not be sent to other chains (because the message-sending
+    /// logic may not have triggered). We can solve this problem by making the message-sending mandatory
+    /// (which we can do once publishers have migrated over).
+    pub prev_publish_time: i64,
+    pub ema_price:         i64,
+    pub ema_conf:          u64,
+}
+
+impl PriceFeedMessage {
+    // The size of the serialized message. Note that this is not the same as the size of the struct
+    // (because of the discriminator & struct padding/alignment).
+    pub const MESSAGE_SIZE: usize = 1 + 32 + 8 + 8 + 4 + 8 + 8 + 8 + 8;
```

Comment:
I still don't really like this MESSAGE_SIZE variable.
You can use size_of() statically : 
`pub fn as_bytes(&self) -> [u8; size_of::<u8>() + size_of::<PriceFeedMessage>()]`

alternatively, add a test in test_sizes.rs

Reply:
well it's not supposed to be the size of the struct (for the reasons noted in the comment). the generative test i added does check that the message exactly fills the array (i.e., no extra bytes left after reading the fields), so this size is tested already.